### PR TITLE
feat(vscode): introduce `autocomplete.strict` config

### PR DESCRIPTION
**Pull request by Simon-He95**

### CODE REVIEW

File: packages/vscode/src/autocomplete.ts [hunk 1]
```diff
@@ -109,6 +109,20 @@ export async function registerAutoComplete(
...
 
+      const offset = window.activeTextEditor!.document.offsetAt(position)
+
+      const afterCode = code.slice(offset)
+        .replace(/"[^"<>{}]*"/g, '')
+        .replace(/\{[^}]*\}/g, '')
+        .replace(/'[^'<>{}]*'/g, '')
+      // get style range
+      const start = code.indexOf('<style')
+      const end = code.lastIndexOf('</style')
```
Comment (antfu) [inline packages/vscode/src/autocomplete.ts:120]: Could we extract this as a function like `shouldProvideAutocomplete` and add some unit test/snapshots to this so we can maintain it easier?

Comment (Simon-He95) [reply packages/vscode/src/autocomplete.ts:120]: sure

**Comment by Simon-He95** [comment]
have some change to commit…

**Comment by Simon-He95** [comment]
done :)

**Pull request merged**